cpp/include/cudf/binaryop.hpp

```diff
@@ -214,5 +214,56 @@ cudf::data_type binary_operation_fixed_point_output_type(binary_operator op,
                                                          cudf::data_type const& lhs,
                                                          cudf::data_type const& rhs);
 
+namespace binops {
+
+/**
+ * @brief Computes output valid mask for op between a column and a scalar
+ */
+rmm::device_buffer scalar_col_valid_mask_and(
+  column_view const& col,
+  scalar const& s,
+  rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+namespace compiled {
+
+/**
+ * @brief Converts scalar to column_view with single element.
+ *
+ * @param scal    scalar to convert
+ * @param stream  CUDA stream used for device memory operations and kernel launches.
+ * @param mr      Device memory resource used to allocate the returned column's device memory
+ * @return        pair with column_view and column containing any auxiliary data to create
+ * column_view from scalar
+ */
+std::pair<column_view, std::unique_ptr<column>> scalar_to_column_view(
+  scalar const& scal,
+  rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
```

Comment:
Due to recent changes, this should now be:
```suggestion
  rmm::cuda_stream_view stream        = cudf::default_stream_value,
```
... and likewise elsewhere in this PR.